holoviews/operation/datashader.py

```diff
@@ -820,13 +820,13 @@ def _process(self, element, key=None):
         (x_range, y_range), (xs, ys), (width, height), (xtype, ytype) = info
         ((x0, x1), (y0, y1)), (xs, ys) = self._dt_transform(x_range, y_range, xs, ys, xtype, ytype)
 
-        df = element.interface.as_dframe(element)
+        df = element.interface.as_dframe(element).copy()
```

Comment:
Is the copy required?

Reply:
Are you trying to sneak in a fix for https://github.com/holoviz/holoviews/issues/5239? :D

(If so, it should probably be done in another PR?)

Reply:
Seems like I did... 

Though, I don't think keeping the fix in this PR is unreasonable. As the changes are related to pandas and datetime.